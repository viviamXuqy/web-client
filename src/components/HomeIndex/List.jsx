import React, { PureComponent } from 'react';
import { object } from 'prop-types';
import { List, Modal } from 'antd';
import { injectIntl } from 'react-intl';

import ResultsCard from '../../shared/containers/ResultsCard';
import ResultDetail from '../../shared/containers/ResultDetail';

class ResultList extends PureComponent {
  static propTypes = {
    intl: object.isRequired,
    results: object.isRequired,
    meta: object.isRequired,
  }

  state = {
    currResult: '', // 当前操作的结果
    modalVisible: false, // 详细信息界面
  }

  showModal = record => {
    this.setState({
      currResult: record,
      modalVisible: true,
    });
  }

  hideModal = () => {
    this.setState({
      modalVisible: false,
    });
  }

  render() {
    const {
      intl: { formatMessage },
      results,
      meta,
    } = this.props;

    const dataSource = [...results.keys()]
      .slice(0, meta.pageSize)
      .map(id => ({
        key: id,
        ...results.get(id),
      }));

    const { currResult } = this.state;

    return (
      <div>
        <div>
          <List
            className="homeindex-results-list"
            grid={{ gutter: 20, column: 5 }}
            dataSource={dataSource}
            renderItem={item => (
              <List.Item className="homeindex-results-list__item">
                <ResultsCard
                  result={item}
                  isResultList={false}
                  showResultDetail={this.showModal}
                />
              </List.Item>
            )}
          />
        </div>
        {this.state.modalVisible &&
          <Modal
            width={790}
            title={formatMessage({ id: 'App.results.resultDetail' })}
            visible={this.state.modalVisible}
            onCancel={this.hideModal}
            footer={null}
          >
            <ResultDetail
              currResult={currResult}
            />
          </Modal>}
      </div>
    );
  }
}

export default injectIntl(ResultList);
