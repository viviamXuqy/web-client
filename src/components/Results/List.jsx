import React, { PureComponent } from 'react';
import { object, func, array } from 'prop-types';
import { List } from 'antd';

import ResultsCard from '../../shared/containers/ResultsCard';

class ResultsList extends PureComponent {
  static propTypes = {
    results: object.isRequired,
    meta: object.isRequired,
    selectedKeys: array.isRequired,
    onPaginationChange: func.isRequired,
    onDelete: func.isRequired,
    onSelect: func.isRequired,
    showResultDetail: func.isRequired,
  }

  isSelected = id => {
    const { selectedKeys } = this.props;
    return selectedKeys.indexOf(id) !== -1;
  }

  render() {
    const {
      results,
      meta,
      onPaginationChange,
      onDelete,
      onSelect,
      showResultDetail,
    } = this.props;

    const dataSource = [...results.keys()]
      .slice(0, meta.pageSize)
      .map(id => ({
        key: id,
        ...results.get(id),
      }));

    return (
      <div>
        <List
          bordered={false}
          className="results-list"
          grid={{ gutter: 20, column: 4 }}
          dataSource={dataSource}
          pagination={{
            ...meta,
            onChange: onPaginationChange,
          }}
          renderItem={item => (
            <List.Item className="results-list__item">
              <ResultsCard
                result={item}
                isResultList
                isSelected={this.isSelected(item.id)}
                onDelete={onDelete}
                onSelect={onSelect}
                showResultDetail={showResultDetail}
              />
            </List.Item>
          )}
        />
      </div>
    );
  }
}

export default ResultsList;
