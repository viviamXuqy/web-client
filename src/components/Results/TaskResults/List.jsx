import React, { PureComponent } from 'react';
import { object, func } from 'prop-types';
import { List, Modal } from 'antd';
import { injectIntl } from 'react-intl';

import TaskResultsCard from '../../../shared/containers/TaskResultsCard';

class TaskResultsList extends PureComponent {
  static propTypes = {
    intl: object.isRequired,
    taskresults: object.isRequired,
    meta: object.isRequired,
    // selectedKeys: array.isRequired,
    onPaginationChange: func.isRequired,
    onDelete: func.isRequired,
    onCardClick: func.isRequired,
  }

  state = {
    previewVisible: false,
    previewImage: null,
  }

  handleImgClick = img => {
    if (img) {
      this.setState({
        previewVisible: true,
        previewImage: img,
      });
    }
  }

  handleCancel = () => {
    this.setState({
      previewVisible: false,
    });
  }

  render() {
    const {
      intl: { formatMessage },
      taskresults,
      meta,
      onPaginationChange,
      onDelete,
      onCardClick,
    } = this.props;

    const dataSource = [...taskresults.keys()]
      .slice(0, meta.pageSize)
      .map(id => ({
        key: id,
        ...taskresults.get(id),
      }));

    const { previewVisible, previewImage } = this.state;

    return (
      <div>
        <div>
          <List
            bordered={false}
            className="taskresults-list"
            grid={{ gutter: 20, column: 4 }}
            dataSource={dataSource}
            pagination={{
              ...meta,
              onChange: onPaginationChange,
            }}
            renderItem={item => (
              <List.Item className="taskresults-list__item">
                <TaskResultsCard
                  taskresult={item}
                  onImgClick={this.handleImgClick}
                  onDelete={onDelete}
                  onCardClick={onCardClick}
                />
              </List.Item>
            )}
          />
        </div>
        <Modal
          visible={previewVisible}
          footer={null}
          onCancel={this.handleCancel}
        >
          <img alt={formatMessage({ id: 'App.control.previewAvatar' })} style={{ width: '100%' }} src={previewImage} />
        </Modal>
      </div>
    );
  }
}

export default injectIntl(TaskResultsList);
