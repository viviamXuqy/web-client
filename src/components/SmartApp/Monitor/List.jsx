import React, { PureComponent } from 'react';
import { object, array, number, func } from 'prop-types';
import { List } from 'antd';
import { injectIntl } from 'react-intl';
import { Scrollbars } from 'react-custom-scrollbars';

import moment from 'moment';

import MonitorCard from '../../../shared/components/MonitorCard';

class MonitorList extends PureComponent {
  static propTypes = {
    type: number.isRequired,
    monitors: array.isRequired,
    intl: object.isRequired,
    onShowMonitorDetail: func.isRequired,
  }

  state = {
    vehicleShowKeys: ['type', 'plate', 'brand', 'carcolor', 'originalTaskName'],
    personShowKeys: ['type', 'sex', 'age', 'originalTaskName'],
  }

  renderThumb = () => {
    const thumbStyle = {
      backgroundColor: '#00D8FF',
      borderRadius: 19,
      width: 5,
    };
    return (
      <div
        style={{ ...thumbStyle }}
      />
    );
  }

  render() {
    const {
      type, monitors, intl, onShowMonitorDetail,
    } = this.props;
    const { personShowKeys, vehicleShowKeys } = this.state;
    const showKeys = type === 1 ? personShowKeys : vehicleShowKeys;
    const dataSource = monitors.map(m => ({
      ...m,
      attri: Object.keys(m.attri).filter(fKey =>
        (showKeys.includes(fKey))).map(key => {
        const classNameVal = key === 'type' ? 'textColorWhite f-bold' : 'textColorWhite';
        return {
          classNameVal,
          name: `${intl.formatMessage({ id: `App.public.struct.${key}` })}: `,
          value: m.attri[key],
        };
      }).concat([{
        classNameVal: 'textColorWhite',
        name: `${intl.formatMessage({ id: 'App.public.struct.time' })}: `,
        value: moment(m.time).format('YYYY/MM/DD hh:mm'),
      }]),
    }));

    return (
      <Scrollbars
        style={{ width: 'calc(100% - 7)', height: 750 }}
        renderThumbVertical={this.renderThumb}
      >
        <div className="monitor__list">
          <List
            itemLayout="horizontal"
            dataSource={dataSource}
            renderItem={item => (
              <List.Item>
                <MonitorCard
                  onShowMonitorDetail={onShowMonitorDetail}
                  monitor={item}
                />
              </List.Item>
            )}
          />
        </div>
      </Scrollbars>
    );
  }
}

export default injectIntl(MonitorList);
