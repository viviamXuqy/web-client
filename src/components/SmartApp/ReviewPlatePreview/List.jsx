import React, { PureComponent } from 'react';
import { array, func } from 'prop-types';
import { List } from 'antd';
import { injectIntl } from 'react-intl';
import { Scrollbars } from 'react-custom-scrollbars';

class ReviewPlatePreviewList extends PureComponent {
  static propTypes = {
    results: array.isRequired,
    onShowResultDetail: func.isRequired,
  }

  state = {
  }

  renderThumb = () => {
    const thumbStyle = {
      backgroundColor: '#00D8FF',
      borderRadius: 19,
      width: 5,
    };
    return (
      <div
        style={{ ...thumbStyle }}
      />
    );
  }

  render() {
    const { results, onShowResultDetail } = this.props;

    return (
      <Scrollbars
        style={{ width: 'calc(100% - 7)', height: 750 }}
        renderThumbVertical={this.renderThumb}
      >
        <div className="monitor__list">
          <List
            itemLayout="horizontal"
            dataSource={results}
            renderItem={item => (
              <List.Item>
                <div className="item">
                  <span>时间：{item.plateCheckedTime}</span>
                  <span>车牌：{item.result ? item.plate.plateNo : '车牌'}</span>
                  <div className="buttons-group">
                    <button onClick={() => onShowResultDetail(item)} className="tablelink" style={{ color: '#00D8FF' }}>查看详情</button>
                  </div>
                </div>
              </List.Item>
            )}
          />
        </div>
      </Scrollbars>
    );
  }
}

export default injectIntl(ReviewPlatePreviewList);
