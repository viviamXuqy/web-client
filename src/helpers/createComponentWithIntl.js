import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line
import LocaleProvider from 'containers/Locale';
import { setStore } from '../utils/storage';
import { nodeWithIntlProp } from './intl-enzyme-test-helper';

const createComponentWithIntl = (children, props = { lang: 'zh-CN' }) => {
  setStore('locale', props.lang);
  return renderer.create(
    <LocaleProvider {...props}>
      {nodeWithIntlProp(children)}
    </LocaleProvider>,
  );
};

export default createComponentWithIntl;
