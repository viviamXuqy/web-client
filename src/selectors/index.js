import { createSelector } from 'reselect';

const taskresultsSelector = state => state.entities.taskresults;
const taskresultsIdsSelector = state => state.entities.taskresultsIds;
export const getTaskResults = createSelector(
  taskresultsIdsSelector,
  taskresultsSelector,
  (ids, taskresults) => {
    const obj = new Map();

    ids.forEach(id => {
      obj.set(id, {
        ...taskresults[id],
        id: `${id}`,
      });
    });

    return obj;
  },
);

const resultsSelector = state => state.entities.results;
const resultsIdsSelector = state => state.entities.resultsIds;
export const getResults = createSelector(
  resultsIdsSelector,
  resultsSelector,
  (ids, results) => {
    const obj = new Map();

    ids.forEach(id => {
      obj.set(id, {
        ...results[id],
        id: `${id}`,
      });
    });

    return obj;
  },
);

const camerasSelector = state => state.entities.cameras;
const camerasIdsSelector = state => state.entities.camerasIds;
export const getCameras = createSelector(
  camerasIdsSelector,
  camerasSelector,
  (ids, cameras) => {
    const obj = new Map();

    ids.forEach(id => {
      obj.set(id, {
        ...cameras[id],
        id: `${id}`,
      });
    });

    return obj;
  },
);

const bayonetsSelector = state => state.entities.bayonets;
const bayonetsIdsSelector = state => state.entities.bayonetsIds;
export const getBayonets = createSelector(
  bayonetsIdsSelector,
  bayonetsSelector,
  (ids, bayonets) => {
    const obj = new Map();

    ids.forEach(id => {
      obj.set(id, {
        ...bayonets[id],
        id: `${id}`,
      });
    });

    return obj;
  },
);

const tasksSelector = state => state.entities.tasks;
const tasksIdsSelector = state => state.entities.tasksIds;
export const getTasks = createSelector(
  tasksIdsSelector,
  tasksSelector,
  (ids, tasks) => {
    const obj = new Map();

    ids.forEach(id => {
      obj.set(id, {
        ...tasks[id],
        id: `${id}`,
      });
    });

    return obj;
  },
);

const logsSelector = state => state.entities.logs;
const logsIdsSelector = state => state.entities.logsIds;
export const getLogs = createSelector(
  logsIdsSelector,
  logsSelector,
  (ids, logs) => {
    const obj = new Map();

    ids.forEach(id => {
      obj.set(id, {
        ...logs[id],
        id: `${id}`,
      });
    });

    return obj;
  },
);

const nvrsSelector = state => state.entities.nvrs;
const nvrsIdsSelector = state => state.entities.nvrsIds;
export const getNvrs = createSelector(
  nvrsIdsSelector,
  nvrsSelector,
  (ids, nvrs) => {
    const obj = new Map();

    ids.forEach(id => {
      obj.set(id, {
        ...nvrs[id],
        id: `${id}`,
      });
    });

    return obj;
  },
);
