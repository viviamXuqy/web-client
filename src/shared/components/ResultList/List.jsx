import React, { PureComponent } from 'react';
import { object, func, array } from 'prop-types';
import { List, Modal } from 'antd';
import { injectIntl } from 'react-intl';

import ResultsCard from '../../../shared/containers/ResultsCard';

class ResultsList extends PureComponent {
  static propTypes = {
    intl: object.isRequired,
    results: object.isRequired,
    meta: object.isRequired,
    selectedKeys: array.isRequired,
    onPaginationChange: func.isRequired,
    onDelete: func.isRequired,
    onSelect: func.isRequired,
    showResultDetail: func.isRequired,
  }

  constructor(props) {
    super(props);
    this.handleImgClick = this.handleImgClick.bind(this);
    this.handleCancel = this.handleCancel.bind(this);
  }

  state = {
    previewVisible: false,
    previewImage: null,
  }

  handleImgClick = img => {
    if (img) {
      this.setState({
        previewVisible: true,
        previewImage: img,
      });
    }
  }

  handleCancel = () => {
    this.setState({
      previewVisible: false,
    });
  }

  isSelected = id => {
    const { selectedKeys } = this.props;
    return selectedKeys.indexOf(id) !== -1;
  }

  render() {
    const {
      intl: { formatMessage },
      results,
      meta,
      onPaginationChange,
      onDelete,
      onSelect,
      showResultDetail,
    } = this.props;

    const dataSource = [...results.keys()]
      .slice(0, meta.pageSize)
      .map(id => ({
        key: id,
        ...results.get(id),
      }));

    const { previewVisible, previewImage } = this.state;

    return (
      <div>
        <div>
          <List
            bordered={false}
            className="results-list2"
            grid={{ gutter: 20, column: 4 }}
            dataSource={dataSource}
            pagination={{
              ...meta,
              onChange: onPaginationChange,
            }}
            renderItem={item => (
              <List.Item className="results-list2__item">
                <ResultsCard
                  result={item}
                  isResultList
                  isSelected={this.isSelected(item.id)}
                  onImgClick={this.handleImgClick}
                  handleCancel={this.handleCancel}
                  onDelete={onDelete}
                  onSelect={onSelect}
                  showResultDetail={showResultDetail}
                />
              </List.Item>
            )}
          />
        </div>
        <Modal
          visible={previewVisible}
          footer={null}
          onCancel={this.handleCancel}
        >
          <img alt={formatMessage({ id: 'App.control.previewAvatar' })} style={{ width: '100%' }} src={previewImage} />
        </Modal>
      </div>
    );
  }
}

export default injectIntl(ResultsList);
