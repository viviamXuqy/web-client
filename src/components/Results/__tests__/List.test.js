import React from 'react';
import createComponentWithIntl from 'helpers/createComponentWithIntl';
import { mountWithIntl } from 'helpers/intl-enzyme-test-helper';

import { getResults } from 'selectors';

import List from 'components/Results/List';

import configureStore from '../../../store/__tests__/configureStore.test';

const { state } = configureStore;

jest.mock('shared/containers/ResultsCard', () => 'mock-ResultsCard');

const onPaginationChange = jest.fn();
const onDelete = jest.fn();
const onSelect = jest.fn();
const showResultDetail = jest.fn();

const initProps = {
  results: getResults(state),
  meta: {
    page: 100000000,
    pageSize: 5,
    total: 100,
  },
  selectedKeys: [],
  onPaginationChange,
  onDelete,
  onSelect,
  showResultDetail,
};

const setup = (props = initProps) => {
  const component = <List {...props} />;
  const componentWithProvider = createComponentWithIntl(component);
  const wrapper = mountWithIntl(component);

  return {
    component,
    componentWithProvider,
    wrapper,
  };
};

describe('List component', () => {
  it('should render without errors', () => {
    const { wrapper } = setup();

    expect(wrapper).toBeTruthy();
  });

  it('should render correct things', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.ant-list.results-list').length).toBe(1);
  });

  it("should render right thing and don't change unexpected", () => {
    const { componentWithProvider } = setup();
    const tree = componentWithProvider.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
