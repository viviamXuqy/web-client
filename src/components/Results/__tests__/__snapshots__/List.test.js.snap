// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render right thing and don't change unexpected 1`] = `
<div>
  <div
    className="ant-list results-list ant-list-split ant-list-grid ant-list-something-after-last-item"
  >
    <div
      className="ant-spin-nested-loading"
      style={null}
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-row"
          style={
            Object {
              "marginLeft": -10,
              "marginRight": -10,
            }
          }
        >
          <div
            className="ant-col-6"
          >
            <div
              className="ant-list-item results-list__item"
              style={
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <mock-ResultsCard
                  isResultList={true}
                  isSelected={false}
                  onDelete={[Function]}
                  onSelect={[Function]}
                  result={
                    Object {
                      "_id": 0,
                      "id": "0",
                      "key": "0",
                      "targetType": "car",
                      "taskId": 0,
                      "taskName": "test",
                      "taskType": 1,
                      "time": 1529355966000,
                    }
                  }
                  showResultDetail={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="ant-col-6"
          >
            <div
              className="ant-list-item results-list__item"
              style={
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <mock-ResultsCard
                  isResultList={true}
                  isSelected={false}
                  onDelete={[Function]}
                  onSelect={[Function]}
                  result={
                    Object {
                      "_id": 1,
                      "id": "1",
                      "key": "1",
                      "targetType": "car",
                      "taskId": 1,
                      "taskName": "test",
                      "taskType": 1,
                      "time": 1529355966000,
                    }
                  }
                  showResultDetail={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="ant-col-6"
          >
            <div
              className="ant-list-item results-list__item"
              style={
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <mock-ResultsCard
                  isResultList={true}
                  isSelected={false}
                  onDelete={[Function]}
                  onSelect={[Function]}
                  result={
                    Object {
                      "_id": 2,
                      "id": "2",
                      "key": "2",
                      "targetType": "car",
                      "taskId": 2,
                      "taskName": "test",
                      "taskType": 1,
                      "time": 1529355966000,
                    }
                  }
                  showResultDetail={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="ant-col-6"
          >
            <div
              className="ant-list-item results-list__item"
              style={
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <mock-ResultsCard
                  isResultList={true}
                  isSelected={false}
                  onDelete={[Function]}
                  onSelect={[Function]}
                  result={
                    Object {
                      "_id": 3,
                      "id": "3",
                      "key": "3",
                      "targetType": "car",
                      "taskId": 3,
                      "taskName": "test",
                      "taskType": 1,
                      "time": 1529355966000,
                    }
                  }
                  showResultDetail={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="ant-col-6"
          >
            <div
              className="ant-list-item results-list__item"
              style={
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <mock-ResultsCard
                  isResultList={true}
                  isSelected={false}
                  onDelete={[Function]}
                  onSelect={[Function]}
                  result={
                    Object {
                      "_id": 4,
                      "id": "4",
                      "key": "4",
                      "targetType": "car",
                      "taskId": 4,
                      "taskName": "test",
                      "taskType": 1,
                      "time": 1529355966000,
                    }
                  }
                  showResultDetail={[Function]}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-list-pagination"
    >
      <ul
        className="ant-pagination "
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-disabled ant-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="上一页"
        >
          <a
            className="ant-pagination-item-link"
          />
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a>
            1
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-2"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={2}
        >
          <a>
            2
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-3"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={3}
        >
          <a>
            3
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-4"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={4}
        >
          <a>
            4
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={5}
        >
          <a>
            5
          </a>
        </li>
        <li
          className="ant-pagination-jump-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="向后 5 页"
        >
          <a
            className="ant-pagination-item-link"
          />
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-20"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={20}
        >
          <a>
            20
          </a>
        </li>
        <li
          aria-disabled={false}
          className=" ant-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={0}
          title="下一页"
        >
          <a
            className="ant-pagination-item-link"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
